GameState.py

```diff
@@ -93,35 +105,58 @@ def getSequence():
 
         return actions
         
-
+    # check if the game over
+    # @return: winner index or -1 if the game has not ended yet
     def isTerminal(self):
         for i in self.players:
+
             if len(i.hand) == 0:
-                return True
-        return False
+                return i
+        return -1
     
+    # check role
     def getUtility(self):
-        if len(self.players[self.toMove()].hand) == 0:
-            return +1
+        currentPlayer = self.players[self.toMove()]
+
+        # if currentplayer has empty hand or the next player has empty hand with same role
+        if len(currentPlayer.hand) == 0 or (len(self.players[nextPlayerIndex].hand) == 0 and self.players[nextPlayerIndex].role == currentPlayer.role):
+            return +100
         
```

Comment:
Should we be defining utility in terms of the player roles (ex: +100 for landlord, -100 for peasants)? This way for expectiminimax, we can have the landlord as max and both peasants as min 

Also this would allow us to determine who won based on the utility of the game state, and we can keep `isTerminal` returning a boolean

Reply:
I think having utilities based on roles might make it easier to ensure that the peasants don't interfere with each other's utilities. We can discuss further when we meet if that would help